core/src/main/java/org/apache/iceberg/jdbc/JdbcClientPool.java

```diff
@@ -43,8 +53,16 @@ public JdbcClientPool(String dbUrl, Map<String, String> props) {
   }
 
   public JdbcClientPool(int poolSize, String dbUrl, Map<String, String> props) {
-    super(poolSize, SQLNonTransientConnectionException.class, true);
+    super(poolSize, SQLTransientException.class, true);
     properties = props;
+    retryableStatusCodes = Sets.newHashSet();
+    retryableStatusCodes.addAll(COMMON_RETRYABLE_CONNECTION_SQL_STATES);
+    String configuredRetryableStatuses = props.get(JdbcUtil.RETRYABLE_STATUS_CODES);
+    if (configuredRetryableStatuses != null) {
+      retryableStatusCodes.addAll(
+          Arrays.stream(configuredRetryableStatuses.split(",")).collect(Collectors.toSet()));
```

Comment:
we might need to make sure that users don't accidentally pass this with spaces: `123, 456, 789`

Reply:
Fixed to handle this case!